defold-spine/src/comp_spine_model.cpp

```diff
@@ -418,10 +418,24 @@ namespace dmSpine
         message.m_Playback    = track.m_Playback;
         message.m_Track       = entry->trackIndex + 1;
 
-        dmGameObject::Result result = dmGameObject::PostDDF(&message, &sender, &receiver, track.m_AnimationCallbackRef, true);
-        if (result != dmGameObject::RESULT_OK)
+        if (track.m_CallbackInfo)
         {
-            dmLogError("Could not send animation_done to listener: %d", result);
+            uint32_t id = track.m_CallbackId;
+            RunTrackCallback(track.m_CallbackInfo, dmGameSystemDDF::SpineAnimationDone::m_DDFDescriptor, (const char*)&message, &sender);
+            // remove after usage only if it's the same callback
+            if (id == track.m_CallbackId)
```

Comment:
If, in a Lua callback, the user calls `spine.play_anim()`, it will destroy the current callback and create a new one (if specified).

Therefore, we need to check whether we are going to remove the same callback that we are running. If not, it has already been removed.

Reply:
I think this would be a good code comment :)

Reply:
it makes sense